prow/plugins/trigger/pr.go

```diff
@@ -49,7 +49,27 @@ func handlePR(c client, trustedOrg string, pr github.PullRequestEvent) error {
 				return fmt.Errorf("could not welcome non-org member %q: %v", author, err)
 			}
 		}
-	case github.PullRequestActionReopened, github.PullRequestActionSynchronize:
+	case github.PullRequestActionReopened:
+		// When a PR is reopened, check that the user is in the org or that an org
+		// member had said "/ok-to-test" before building.
+		comments, err := c.GitHubClient.ListIssueComments(pr.PullRequest.Base.Repo.Owner.Login, pr.PullRequest.Base.Repo.Name, pr.PullRequest.Number)
```

Comment:
Why don't we do the self-membership check first and potentially avoid this list plus any commentors auth checks?

Reply:
We need to list comments to clean up stale comments for the case in which the PR author becomes an org member.
Personally I don't think cleaning up stale comments in this case is worth the extra API tokens so I opened #5382.

Reply:
@cjwagner agreed. Maybe we don't care as much though because reopening PRs is not that common? We should start tracking stuff like that in metrics.

Reply:
Yeah I was going to leave the reopen logic as is so that reopening a PR can be used to clean up comments and such. Just changing the behavior for sync events should save a lot of tokens.

I believe @Q-Lee is doing some work to get us metrics on webhook event frequency for different webhook types.